runtime/compiler/net/ServerStream.cpp

```diff
@@ -0,0 +1,312 @@
+   int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+   if (sockfd < 0)
+      {
+      perror("can't open server socket");
+      exit(1);
+      }
+
+   // see `man 7 socket` for option explanations
+   int flag = true;
+   if (setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, (void *)&flag, sizeof(flag)) < 0)
+      {
+      perror("Can't set SO_REUSEADDR");
+      exit(-1);
+      }
+   if (setsockopt(sockfd, SOL_SOCKET, SO_KEEPALIVE, (void *)&flag, sizeof(flag)) < 0)
+      {
+      perror("Can't set SO_KEEPALIVE");
+      exit(-1);
+      }
+
+   struct timeval timeoutForListening = {0, 0}; // Wait forever
+   if (setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO, (void *)&timeoutForListening, sizeof(timeoutForListening)) < 0)
+      {
+      perror("Can't set option SO_RCVTIMEO on sockfd socket");
+      exit(-1);
+      }
+   if (setsockopt(sockfd, SOL_SOCKET, SO_SNDTIMEO, (void *)&timeoutForListening, sizeof(timeoutForListening)) < 0)
+      {
+      perror("Can't set option SO_SNDTIMEO on sockfd socket");
+      exit(-1);
+      }
+
+   struct sockaddr_in serv_addr;
+   memset((char *)&serv_addr, 0, sizeof(serv_addr));
+   serv_addr.sin_family = AF_INET;
+   serv_addr.sin_addr.s_addr = htonl(INADDR_ANY);
+   serv_addr.sin_port = htons(port);
+
+   if (bind(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0)
+      {
+      perror("can't bind server address");
+      exit(1);
+      }
+   if (listen(sockfd, SOMAXCONN) < 0)
+      {
+      perror("listen failed");
+      exit(1);
+      }
+
+   while (true)
+      {
+      struct sockaddr_in cli_addr;
+      socklen_t clilen = sizeof(cli_addr);
+
+      int connfd = accept(sockfd, (struct sockaddr *)&cli_addr, &clilen);
+      if (connfd < 0)
+         {
+         if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "Error accepting connection: errno=%d", errno);
+         continue;
+         }
+
+      struct timeval timeoutMsForConnection = {timeoutMs/1000, timeoutMs*1000%1000000};
+      if (setsockopt(connfd, SOL_SOCKET, SO_RCVTIMEO, (void *)&timeoutMsForConnection, sizeof(timeoutMsForConnection)) < 0)
+         {
+         perror("Can't set option SO_RCVTIMEO on connfd socket");
+         exit(-1);
+         }
+      if (setsockopt(connfd, SOL_SOCKET, SO_SNDTIMEO, (void *)&timeoutMsForConnection, sizeof(timeoutMsForConnection)) < 0)
+         {
+         perror("Can't set option SO_SNDTIMEO on connfd socket");
+         exit(-1);
+         }
+
+#if defined(JITSERVER_ENABLE_SSL)
+      BIO *bio = NULL;
+      if (sslCtx && !acceptOpenSSLConnection(sslCtx, connfd, bio))
+         continue;
+
+      ServerStream *stream = new (PERSISTENT_NEW) ServerStream(connfd, bio);
+#else
+      ServerStream *stream = new (PERSISTENT_NEW) ServerStream(connfd);
+#endif
+      compiler->compile(stream);
+      }
+
+   // TODO This will never be called - if the server actually shuts down properly then we should do this
```

Comment:
Rephrase ro eliminate the word TODO.
" The following piece of code will be executed only of the server shuts down properly"